tofu/imas2tofu/_core.py

```diff
@@ -2988,13 +3086,129 @@ def _checkformat_Cam_geom(self, ids=None, geomcls=None, indch=None):
 
         lgeom = [kk for kk in dir(tfg) if 'Cam' in kk]
         if geomcls not in [False] + lgeom:
-            msg = "Arg geomcls must be in %s"%str([False]+lgeom)
+            msg = "Arg geomcls must be in {}".format([False]+lgeom)
             raise Exception(msg)
 
         return geomcls
 
+    def _get_indch_geomtdata(self, indch=None, indch_auto=None,
```

Comment:
hidden method created to handle automated determination of indch in a modular way